modules/swagger-codegen/src/main/resources/Java/modelEnum.mustache

```diff
@@ -60,7 +60,7 @@ public enum {{#datatypeWithEnum}}{{{.}}}{{/datatypeWithEnum}}{{^datatypeWithEnum
 
     @Override
     public {{#datatypeWithEnum}}{{{.}}}{{/datatypeWithEnum}}{{^datatypeWithEnum}}{{{classname}}}{{/datatypeWithEnum}} read(final JsonReader jsonReader) throws IOException {
-      {{{dataType}}} value = jsonReader.{{#isInteger}}nextInt(){{/isInteger}}{{^isInteger}}next{{{dataType}}}(){{/isInteger}};
+      {{{dataType}}} value = {{#isFloat}}new Float(jsonReader.nextDouble()); // floats shall be restricted so this cast is safe{{/isFloat}}{{^isFloat}}jsonReader.{{#isInteger}}nextInt(){{/isInteger}}{{^isInteger}}next{{{dataType}}}(){{/isInteger}};{{/isFloat}}
```

Comment:
Can we overflow here while converting double => float?

Reply:
Yes. ```new Float(Double.MAX_VALUE)``` will result in ```Infinity```.

That is not the best solution.

To make it better the JsonReader functionality has to be approximately reimplemented. Something like this:
```Java
    final double nextDouble = jsonReader.nextDouble();
    float result = (float) nextDouble;
    if (Float.isInfinite(result)) {
      throw new MalformedJsonException("JSON forbids infinities: " + result + " for " + nextDouble);
    }
```
(Compare with: https://github.com/google/gson/blob/b1fb9ca9a1bea5440bc6a5b506ccf67236b08243/gson/src/main/java/com/google/gson/stream/JsonReader.java#L909)

But this seems like overkill.


My thoughts were:
In case a float value is restricted by enumerations this method will result in ```null``` if the enum value can't be matched. If a decimal number is to big for float, leading to ```Infinity```, the value can't be matched and the result will be ```null```, too.

Reply:
> But this seems like overkill.

ok, seems reasonable.